src/components/dialog/Dialog.tsx

```diff
@@ -0,0 +1,39 @@
+// The dialog element allready has keyboard support
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React, { forwardRef } from "react";
+import iconCross from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/basic/icon-cross.svg";
+import useEscapeKey from "./useEscapeKey";
+
+type DialogType = {
+  children: React.ReactNode;
+  closeDialog: () => void;
+};
+
+const Dialog = forwardRef<HTMLDialogElement, DialogType>(
+  ({ children, closeDialog }, ref) => {
+    useEscapeKey({ closeDialog });
+    return (
+      <dialog
+        className="dialog"
+        ref={ref}
+        // Close dialog when clicking outside of it (::backdrop pseudo-element)
+        onClick={({ currentTarget, target }) => {
+          if (currentTarget === target) {
+            closeDialog();
+          }
+        }}
+      >
+        <button
+          type="button"
+          onClick={closeDialog}
+          className="dialog__close-button"
+        >
+          <img src={iconCross} alt="close icon" />
```

Comment:
Please provide a proper alt text.

Close icon is not suitable. We might already have something usable elsewhere. You could also make the `alt` text empty and provide an `aria-label` with text for the button.